shared/src/commonMain/kotlin/com/example/travelapp_kmp/listing/ListScreenState.kt

```diff
@@ -1,12 +1,34 @@
 package com.example.travelapp_kmp.listing
 
+import org.jetbrains.compose.resources.DrawableResource
+import org.jetbrains.compose.resources.ExperimentalResourceApi
+
 
 sealed interface ListScreenState {
     object Loading : ListScreenState
     data class Error(val message: String) : ListScreenState
     data class Success(
         val countriesList: List<Country>,
-        val selectedCountry: Country,
-        val selectedItemIndex: Int = 0,
-    ) : ListScreenState
+        val selectedTouristPlacesIndex: Int = 0,
+        val selectedCountryIndex: Int = 0,
+    ) : ListScreenState {
+        fun getWeather(): Weather? {
+            return countriesList[selectedCountryIndex].touristPlaces[selectedTouristPlacesIndex].weather
+        }
+
+        val countriesTouristPlaces: List<TouristPlace> =
+            countriesList[selectedCountryIndex].touristPlaces
+
+        val nameTouristPlaceSelected: String =
+            countriesList[selectedCountryIndex].touristPlaces[selectedTouristPlacesIndex].name
+
+        val nameCountrySelected: String = countriesList[selectedCountryIndex].name
+
+        @OptIn(ExperimentalResourceApi::class)
+        fun getImagePlaceholder(): DrawableResource {
+            return countriesList[selectedCountryIndex].touristPlaces[selectedTouristPlacesIndex].images[0]
+        }
```

Comment:
That can be move to the tourist place model data class

Reply:
Yes but need pass index selected and show put here no problem and any suggestion??? @SEAbdulbasit